packages/client-sdk-nodejs/src/internal/internal-auth-client.ts

```diff
@@ -28,18 +29,38 @@ import {
   AllTopics,
   isCacheName,
   isTopicName,
+  GenerateDisposableToken,
+  AllItems,
+  isCacheItemKey,
+  isCacheItemKeyPrefix,
 } from '@gomomento/sdk-core';
 import {IAuthClient} from '@gomomento/sdk-core/dist/src/internal/clients';
 import {AuthClientProps} from '../auth-client-props';
 import {normalizeSdkError} from '@gomomento/sdk-core/dist/src/errors';
 import {
+  TemporaryTokenScope,
   asCachePermission,
   asPermissionsObject,
   asTopicPermission,
   isCachePermission,
   isPermissionsObject,
   isTopicPermission,
+  PredefinedScope,
+  isTemporaryTokenPermissionsObject,
+  asTemporaryTokenPermissionsObject,
+  TemporaryTokenCachePermission,
+  isTemporaryTokenCachePermission,
+  asTemporaryTokenCachePermission,
 } from '@gomomento/sdk-core/dist/src/auth/tokens/token-scope';
+import {permission_messages} from '@gomomento/generated-types/dist/permissionmessages';
+
+const textEncoder = new TextEncoder();
+function convert(v: string | Uint8Array): Uint8Array {
+  if (typeof v === 'string') {
+    return textEncoder.encode(v);
+  }
+  return v;
+}
```

Comment:
this was not very gracefully put in here, but the CacheItemSelector accepts only Uint8Array, not string